File: Tests/iOS Tests/PNTimeTokenTests.m
//
//  PNTimeTokenTests.m
//  PubNub Tests
//
//  Created by Jordan Zucker on 6/28/15.
//
//

#import <UIKit/UIKit.h>
#import <XCTest/XCTest.h>
#import <PubNub/PubNub.h>

#import "PNBasicSubscribeTestCase.h"

@interface PNTimeTokenTests : PNBasicSubscribeTestCase

@end

@implementation PNTimeTokenTests

- (BOOL)isRecording{
    return NO;
}

- (void)testTimeToken {
    XCTestExpectation *timeTokenExpectation = [self expectationWithDescription:@"timeToken"];
    [self.client timeWithCompletion:^(PNTimeResult *result, PNErrorStatus *status) {
        XCTAssertNil(status);
        XCTAssertNotNil(result);
        XCTAssertEqual(result.operation, PNTimeOperation);
        XCTAssertEqual(result.statusCode, 200);
        XCTAssertEqualObjects(result.data.timetoken, @14355553745683928);
        [timeTokenExpectation fulfill];
    }];
    [self waitForExpectationsWithTimeout:5 handler:^(NSError *error) {
        XCTAssertNil(error);
    }];
}

@end

File: Tests/iOS Tests-Swift/PNTimeTokenTests.swift
//
//  PNTimeTokenTests.swift
//  PubNub Tests
//
//  Created by Jordan Zucker on 7/28/15.
//
//

import UIKit
import XCTest

class PNTimeTokenTests: PNBasicSubscribeTestCase {
    override func isRecording() -> Bool {
        return false
    }

    func testTimeToken() {
        // This is an example of a functional test case.
        let testTokenExpectation = self.expectationWithDescription("timeToken")
        self.client.timeWithCompletion { (result, status) -> Void in
            XCTAssertNil(status)
            XCTAssertNotNil(result)
            XCTAssertEqual(result!.operation, PNOperationType.TimeOperation)
            XCTAssertEqual(result!.statusCode, 200)
            XCTAssertEqual(result!.data.timetoken, NSNumber(longLong: 14355553745683928))
            testTokenExpectation.fulfill()
        }
        self.waitForExpectationsWithTimeout(5, handler: { (error) -> Void in
            XCTAssertNil(error, "Encountered error with time token test")
        })
    }
}
